op-supervisor/supervisor/types/types.go

```diff
@@ -1,19 +1,52 @@
 package types
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/holiman/uint256"
 )
 
 type Identifier struct {
+	Origin      common.Address
+	BlockNumber uint64
+	LogIndex    uint64
+	Timestamp   uint64
+	ChainID     *uint256.Int
```

Comment:
I intentionally made this embedded, not a pointer, so identifiers are safe and easy to use as map key and copied easily via struct-shallow copy. Will change this back, otherwise looks good to me. Thanks!